testsupport/wait/host.go

```diff
@@ -1186,13 +1187,28 @@ func (a *HostAwaitility) WaitForNotificationWithName(t *testing.T, notificationN
 			return false, fmt.Errorf("notification found with name does not have the expected type")
 		}
 
-		return matchNotificationWaitCriterion([]toolchainv1alpha1.Notification{notification}, criteria...), nil
+		return matchNotificationWaitCriterion([]toolchainv1alpha1.Notification{*notification}, criteria...), nil
 	})
 	// no match found, print the diffs
 	if err != nil {
-		a.printNotificationWaitCriterionDiffs(t, []toolchainv1alpha1.Notification{notification}, criteria...)
+		a.printNotificationWaitCriterionDiffs(t, []toolchainv1alpha1.Notification{*notification}, criteria...)
 	}
-	return notification, err
+	return *notification, err
+}
+
+// WaitForNotificationToBeNotCreated waits and checks that notification is NOT created.
+func (a *HostAwaitility) WaitForNotificationToBeNotCreated(t *testing.T, notificationName string) error {
+	t.Logf("waiting to check notification with name '%s' is NOT created", notificationName)
+	return wait.Poll(a.RetryInterval, 10*time.Second, func() (done bool, err error) {
+		notification := &toolchainv1alpha1.Notification{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: notificationName, Namespace: a.Namespace}, notification); err != nil {
+			if errors.IsNotFound(err) {
+				return true, nil
+			}
```

Comment:
I'm afraid that this won't wait for 10s, but will exit at the first run when it doesn't find anything.
The expected behavior is that the poll will be running for the whole 10s and only if it finds the notification that it should immediately exit. In other words, you need to swap the returned values and the expected outcome (if you want to stay with wait.Pool).

Reply:
Yes right, Thanks for pointing that. i updated it now